lib/publiccloud/provider.pm

```diff
@@ -126,6 +121,89 @@ sub parse_ipa_output {
     return $ret;
 }
 
+=head2 create_ssh_key
 
-1;
+Creates an ssh keypair in a given file path by $args{ssh_private_key_file}
+
+=cut
+sub create_ssh_key {
+    my ($self, %args) = @_;
+    if (script_run('test -f ' . $args{ssh_private_key_file}) != 0) {
+        assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; mkdir -p $SSH_DIR');
+        assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -f ' . $args{ssh_private_key_file});
+    }
+}
+
+=head2 run_ipa
+
+called by childs within ipa function
+
+=cut
+sub run_ipa {
+    my ($self, %args) = @_;
+    $args{cleanup}              //= 1;
+    $args{ssh_private_key_file} //= '.ssh/id_rsa';
+    $args{tests}                //= '';
+    $args{timeout}              //= 60 * 20;
+    $args{results_dir}          //= 'ipa_results';
+    $args{distro}               //= 'sles';
+    $args{tests} =~ s/,/ /g;
+
+    $self->create_ssh_key(ssh_private_key_file => $args{ssh_private_key_file}) if (!-e $args{ssh_private_key_file});
```

Comment:
Sorry, `-e ` was wrong. It checks the existence of the file in worker...
I just saw that provider::create_ssh_key() is checking this anyway.

Reply:
ok, I will remove the if condition

Reply:
ok, after testing it, it's not possible to remove the if condition because we are overwritting the argument for EC2, which is set to $self->ssh_key_file;

Reply:
I think the `if` condition makes no sense. It check a file  somewhere on the filesystem of the worker.
We override the create_ssh_key() function in `publiccloud::ec2`, so  the ssh-keygen isn't called for ec2.